messagecatalog/spec.md

```diff
@@ -0,0 +1,1475 @@
+  detail than the `description`. The text MAY use [CommonMark](https://commonmark.org/) markdown formatting.
+- Constraints:
+  - OPTIONAL
+
+#### `tags` (message definition group)
+
+- Type: `String`
+- Description: A single string containing a comma-separated list of tags
+  describing the message definition group for search indexing.
+- Constraints:
+  - OPTIONAL
+- Examples:
+  - "exampleapp,examplecorp"
+
+#### `createdtimeutc` (message definition group)
+
+- Type: `Timestamp`
+- Description: Instant when the message definition group was added to the
+  catalog.
+- Constraints:
+  - OPTIONAL
+- Assigned by the server.
+
+#### `updatedtimeutc` (message definition group)
+
+- Type: `Timestamp`
+- Description: Instant when the message definition group was last updated
+- Constraints:
+  - OPTIONAL
+- Assigned by the server.
+
+#### `definitions' (message definition group)
+
+- Type: List of [message definitions](#22-message-definition-and-message-definition-version)
+- Description: List of message definitions belonging to this group
+
+## 2.2. Message definition and message definition version
+
+A message definition is a description of a message or event data structure of a
+certain message format. This specification provides message formats for
+[CloudEvents](#32-cncf-cloudevents), [AMQP](#33-oasis-amqp),
+[MQTT](#34-oasis-mqtt), and [NATS](#35-cncf-nats). Extension formats are
+permitted. A message definition MAY have multiple versions.
+
+The "message definition" object contains all information that MUST be identical
+or is common across all versions of such a definition. Specifically `id` and
+`format` MUST NOT be overridden by versions added later. The `documentation`
+field is common and expected to cover all versions including explanation of
+differences and changes between versions.
+
+Each "message definition version" inherits those attribute values and adds elements
+that MAY vary across versions. Each "message definition" MUST have at least one
+"message definition version".
+
+When a _message definition version_ document is retrieved or posted singly, the
+attributes of the message definition become part of its representation as if it
+were a single object. When a _message definition_ document is retrieved or
+posted with some or all of its versions, the versions are held in an array named
+`versions` and all common attributes are held in the containing object.
+
+The example below, illustrating a CloudEvent without payload for brevity, shows an
+example of a _message definition version_ document:
+
+```JSON
+{
+    "id" : "io.cloudevents.messagecatalog.deleted",
+    "authority" : "cloudevents.io",
+    "description" : "Raised when an object has been deleted from the catalog",
+    "format" : "CloudEvents",
+    "versions" : [
+      {
+        "version" : 1,
+        "definition" : {
+          "attributes" : {
+            "specversion" : {
+              "value" :"1.0",
+              "required" : true,
```

Comment:
this seems fairly redundant if the format is already defined as cloudevents, we should just need to list core attributes that are expected and possibly extensions that are likely?

Reply:
We’ve talked about how there might be a single doc with all data from all our specs inlined, I wonder if it would be useful to show an example of that doc in an upcoming call so that people can see whether there really is duplication of data.